src/profile/mfa/mfa_factor_sources/security_questions_factor_source/security_questions_sealed_mnemonic.rs

```diff
@@ -0,0 +1,132 @@
+use crate::prelude::*;
+
+/// ❗️ NOT PRODUCTION READY YET ❗️
+/// A mnemonic encrypted by answers to security questions
+/// ❗️ NOT PRODUCTION READY YET ❗️
+#[derive(
+    Serialize, Deserialize, Clone, PartialEq, Eq, Hash, Debug, uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+#[allow(non_camel_case_types)]
+pub struct SecurityQuestionsSealed_NOT_PRODUCTION_READY_Mnemonic {
+    pub security_questions: Security_NOT_PRODUCTION_READY_Questions,
+
+    /// A versioned Key Derivation Function (KDF) algorithm used to produce a set
+    /// of Encryption keys from a set of security questions and answers
+    pub kdf_scheme: SecurityQuestions_NOT_PRODUCTION_READY_KDFScheme,
+
+    /// The scheme used to encrypt the Security Questions factor source
+    /// mnemonic using one combination of answers to questions, one of many.
+    pub encryption_scheme: EncryptionScheme,
+
+    /// The N many encryptions of the mnemonic, where N corresponds to the number of derived keys
+    /// from the `keyDerivationScheme`
+    pub encryptions: Vec<Exactly60Bytes>, // FIXME: Set?
+}
+
+impl SecurityQuestionsSealed_NOT_PRODUCTION_READY_Mnemonic {
+    pub const QUESTION_COUNT: usize = 6;
+    pub fn new_by_encrypting(
+        mnemonic: Mnemonic,
+        with: Security_NOT_PRODUCTION_READY_QuestionsAndAnswers,
+        kdf_scheme: SecurityQuestions_NOT_PRODUCTION_READY_KDFScheme,
+        encryption_scheme: EncryptionScheme,
+    ) -> Result<Self> {
+        let questions_and_answers = with;
+        if questions_and_answers.len() != Self::QUESTION_COUNT {
+            return Err(CommonError::InvalidQuestionsAndAnswersCount {
+                expected: Self::QUESTION_COUNT as u16,
+                found: questions_and_answers.len() as u16,
+            });
+        }
+        let security_questions = questions_and_answers
+            .iter()
+            .map(|qa| qa.question.clone())
+            .collect::<Security_NOT_PRODUCTION_READY_Questions>();
+
+        let mnemonic_entropy = Exactly32Bytes::try_from(mnemonic.to_entropy().as_ref()).expect("SecurityQuestionsFactorSource mnemonics SHOULD ALWAYS be 32 bytes of entropy.");
+
+        let encryption_keys = kdf_scheme.derive_encryption_keys_from_questions_and_answers(questions_and_answers).expect("TODO validate that answer is non-empty BEFORE passing it here.");
+
+        let encryptions = encryption_keys
+            .into_iter()
+            .map(|k| {
+                encryption_scheme.encrypt(mnemonic_entropy, &mut k.clone())
+            })
+            .map(|vec| {
+                Exactly60Bytes::try_from(vec)
+                    .expect("Should have been 60 bytes")
+            })
+            .collect_vec();
+
+        Ok(Self {
```

Comment:
just want to double check if I understand the procedure correctly. We multiple encrypted contents of a mnemonic (to allow for some errors in Q & A set), and then when we decrypt and generate multiple decryption keys, we try to decrypt until we succeed?

Reply:
I think you understand it correctly, but your description was a bit inaccurate/unclear:

The word "generate" entails randomness, so it is inaccurate (we say we **generate** entropy, from which we encode a BIP39 mnemonic). What we do is we "derive" Symmetric (En/De)cryption keys from set of pairs of (Question, Answer) tuples. "Derive" is not random, it is determinisitc.

We derive the **same encryption keys** part of *creation* (=encryption) *and usage* (=decryption) of a of a SecurityQuestionsFactorSource. Or well, the same if not too many questions were incorrectly answered.

And yes we try all combinations, 4*4 = 16, of encryptions of mnemonic and decryption keys, the first combination to succeed => return the decrypted mnemonic.